src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/util/SalesforceSplitUtil.java

```diff
@@ -105,31 +115,64 @@ private static BatchInfo[] getBatches(String query, BulkConnection bulkConnectio
    * @throws IOException       failed to close the query
    */
   private static BatchInfo[] runBulkQuery(BulkConnection bulkConnection, String query,
-                                          boolean enablePKChunk, String operation)
-    throws AsyncApiException, IOException {
+                                          boolean enablePKChunk, String operation,
+                                          Long initialRetryDuration, Long maxRetryDuration,
+                                          Integer maxRetryCount, Boolean retryOnBackendError)
+    throws AsyncApiException, IOException, InterruptedException {
 
     SObjectDescriptor sObjectDescriptor = SObjectDescriptor.fromQuery(query);
     JobInfo job = SalesforceBulkUtil.createJob(bulkConnection, sObjectDescriptor.getName(), getOperationEnum(operation),
-                                               null, ConcurrencyMode.Parallel, ContentType.CSV);
-    BatchInfo batchInfo;
-    try (ByteArrayInputStream bout = new ByteArrayInputStream(query.getBytes())) {
-      batchInfo = bulkConnection.createBatchFromStream(job, bout);
+      null, ConcurrencyMode.Parallel, ContentType.CSV);
+    final BatchInfo batchInfo;
+    try {
+      if (retryOnBackendError) {
```

Comment:
instead of having if else for whole code, if !retryEnabled, set maxretryattempts to 0 at all places

Reply:
done

Reply:
change this at all places